datacube_stats/utils/__init__.py

```diff
@@ -96,6 +96,53 @@ def to_float(da):
     return data.apply(to_float, keep_attrs=True)
 
 
+def cast_back(data, measurements):
+    """
+    Cast calculated statistic `Dataset` into intended data types.
+    When going through intermediate representation as floats,
+    restore `nodata` values in place of `NaN`s.
+    """
+    assert isinstance(data, xarray.Dataset)
+    measurements = {measurement['name']: measurement
+                    for measurement in measurements}
+
+    data_vars = [name for name in data.data_vars]
+    assert set(data_vars) == set(measurements.keys())
+
+    def cast(da):
+        """ Cast `DataArray` into intended type. """
+        output_measurement = measurements[da.name]
+        expected_dtype = np.dtype(output_measurement['dtype'])
+        actual_dtype = da.dtype
+
+        # our intermediate data type
+        float32 = np.dtype(np.float32)
```

Comment:
pretty sure `np.float32` is already a dtype object.

Reply:
```python
isinstance(numpy.float32, numpy.dtype) == False
type(numpy.float32) == type
# although
numpy.float32 == numpy.dtype('f4')
```

Reply:
ok